src/interface/viewport.c

```diff
@@ -1805,9 +1805,9 @@ void viewport_paint_setup()
 		.y = (dpi->y - 16) & 0xFFE0
 	};
 
-	sint16 half_x = mapTile.x / 2;
+	sint16 half_x = mapTile.x >> 1;
```

Comment:
This is implementation defined behaviour this was the reason why it's divide and not right shift.